pkg/mimir/mimir_test.go

```diff
@@ -151,7 +151,7 @@ func TestMimir(t *testing.T) {
 		Target: []string{All, AlertManager},
 	}
 
-	c, err := New(cfg)
+	c, err := New(cfg, prometheus.NewPedanticRegistry())
```

Comment:
If we're not checking metrics, we should use `nil` to make it clear.

Reply:
See: https://github.com/grafana/mimir/pull/2750#discussion_r947795562

Reply:
Addressed in https://github.com/grafana/mimir/pull/2750/commits/ae2a9b3eb1b476ea03bf96c6f653bd07121b9b33

Reply:
Unfortunately I had to rollback this because if registry is `nil` then `weaveworks/common/server` uses the default one.